sarracen/interpolate/interpolate.py

```diff
@@ -447,16 +466,19 @@ def interpolate_2d_cross(data: 'SarracenDataFrame', target: str, x: str = None,
     if y2 == y1 and x2 == x1:
         raise ValueError('Zero length cross section!')
 
+    mass_data = _get_mass(data)
+    rho_data = _get_density(data)
+    w_data = data[target].to_numpy() * mass_data / rho_data
+
```

Comment:
There needs to be a check for if 'target' is density, right? Similar to interpolate_3d(), lines 550-555